sycl/source/detail/usm/usm_impl.cpp

```diff
@@ -41,7 +41,8 @@ void *alignedAlloc(size_t Alignment, size_t Size, const context &Ctxt,
   }
   }
 
-  PI_CHECK(Error);
+  // Error is for debugging purposes.
+  // The spec wants a nullptr returned, not an exception.
```

Comment:
So should we add something like this?
```c++
if (Error != PI_SUCCESS) return nullptr;
```

@smaslov-intel, don't we have some helper function similar to `PI_CHECK` logging errors without throwing an exception?

Reply:
There is PI_CHECK_THROW to throw an excpetion diffferent from cl::sycl::runtime_error, which is thrown in case of an error by PI_CHECK. We could add PI_CHECK_NOTHROW, but in this case someone needs to return null manually anyway.

Reply:
`RetVal` initial value is replaced by `Dispatch->hostMemAlloc` return value.
Does this function return `nullptr` for case where resulting `Error != PI_SUCCESS`?